bundles/core/src/main/java/com/adobe/cq/commerce/core/components/services/urls/UrlProvider.java

```diff
@@ -172,9 +219,10 @@
      * configuration for product page URLs.
      *
      * @param request The current Sling HTTP request.
+     * @param page The current page
      * @return the parsed {@link ProductUrlFormat.Params}
      */
-    ProductUrlFormat.Params parseProductUrlFormatParameters(SlingHttpServletRequest request);
+    ProductUrlFormat.Params parseProductUrlFormatParameters(SlingHttpServletRequest request, Page page);
```

Comment:
Those changes to the API are unnecessary and not required. The context should be fully given by the request.